middlewares/secrets.go

```diff
@@ -0,0 +1,32 @@
+package middlewares
+
+import (
+	"net/http"
+)
+
+type SecretsConfig struct {
+	OmitHeaders []string `yaml:"omit_headers"` // Headers to be omitted for secrets protection
+}
+
+// Secrets middleware removes specified headers from the request and response to enhance security.
+func NewSecretsMiddleware(config SecretsConfig) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// Omit secrets from request
+			for _, header := range config.OmitHeaders {
```

Comment:
I think we don't have any problem with headers sent, only omit from response

Reply:
Fixed